### PR TITLE
feat(unlock-app) - checkout config to event page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
Set checkout config for a specific event as part of https://github.com/unlock-protocol/unlock/issues/11019 

This PR needs 

**Overview** 

Screenshot 2023-05-31 at 16 52 16

**Disabled check config** 

Screenshot 2023-05-31 at 16 53 24

**Set check config** 

Screenshot 2023-05-31 at 16 53 45

**Checkout config missing (lockManager)**  

Screenshot 2023-05-30 at 13 29 04

**Checkout config missing (non-lockManager)**  

Screenshot 2023-05-30 at 13 29 14

**Registration options based on checkout config** 

Screenshot 2023-05-31 at 16 45 47

**Registration modal **
 
Screenshot 2023-05-30 at 15 39 36

**Verifiers (multiple locks in checkout config)**
 
Screenshot 2023-05-31 at 16 34 03

**Verifiers (multiple locks in checkout config)**

Screenshot 2023-05-31 at 16 55 59

**Mobile** 
Screenshot 2023-05-31 at 16 47 00

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by searchableguy — APPROVED** [review]

File: unlock-app/src/components/content/event/EventCheckoutUrl.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                            Your configuration is not present? you can create
+                            one from{' '}
+                            <Link
+                              className="font-semibold text-brand-ui-primary"
+                              href={checkoutUrl}
+                              rel="noopener noreferrer"
+                            >
+                              Create a checkout URL
+                            </Link>{' '}
+                            .
```
Comment (kalidiagne) [inline unlock-app/src/components/content/event/EventCheckoutUrl.tsx:201]: @julien51 can we improve copy here?

File: unlock-app/src/components/content/event/EventDetails.tsx [hunk 2]
```diff
@@ -167,6 +171,168 @@ const CoverImageDrawer = ({
...
+
+  // not match found for the assigned Checkout ID, for example could be deleted
+  if (!checkoutConfig || locks.length === 0) {
+    return (
+      <Card className="grid gap-6 mt-10 lg:mt-0">
+        <span className="text-2xl font-bold text-gray-900">Registration</span>
+        <span>
+          {isManager
+            ? 'Your Checkout URL assigned to this event is deleted or not valid. Please make sure assign an existing one.'
+            : 'Registration details is not configured.'}
```
Comment (kalidiagne) [inline unlock-app/src/components/content/event/EventDetails.tsx:226]: @julien51 can we improve copy here?

File: unlock-app/src/components/content/event/EventDetails.tsx [hunk 3]
```diff
@@ -536,11 +701,26 @@ export const EventDetails = ({ lockAddress, network }: EventDetailsProps) => {
...
+                    disabled={!isLockManager}
+                  />
+                </Disclosure>
+              )}
+
               <Disclosure
-                label="Verifiers"
-                description="Add & manage trusted users at the event to help check-in attendees"
+                label="Checkout URL"
+                description="Select one of their saved checkout URL that will be used for this event."
```
Comment (kalidiagne) [inline unlock-app/src/components/content/event/EventDetails.tsx:721]: @julien51 can we improve copy here?

File: unlock-app/src/components/content/event/EventCheckoutUrl.tsx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+  if (!hasConfigList) {
+    return (
+      <span>
+        No items to choose from.{' '}
+        <Link
+          className="font-semibold text-brand-ui-primary"
+          href={checkoutUrl}
+          rel="noopener noreferrer"
+        >
+          Create a checkout URL
```
Comment (julien51) [inline unlock-app/src/components/content/event/EventCheckoutUrl.tsx:140]:
```suggestion
          Create a new checkout URL, where you can configure the checkout steps for your event.
```

File: unlock-app/src/components/content/event/EventCheckoutUrl.tsx [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                        description={
+                          <span>
+                            Your configuration is not present? you can create
+                            one from{' '}
+                            <Link
+                              className="font-semibold text-brand-ui-primary"
+                              href={checkoutUrl}
+                              rel="noopener noreferrer"
+                            >
+                              Create a checkout URL
```
Comment (julien51) [inline unlock-app/src/components/content/event/EventCheckoutUrl.tsx:218]:
```suggestion
                              create a new one
```

File: unlock-app/src/components/content/event/EventDetails.tsx [hunk 6]
```diff
@@ -167,6 +171,168 @@ const CoverImageDrawer = ({
...
+  }
+
+  // not match found for the assigned Checkout ID, for example could be deleted
+  if (!checkoutConfig || locks.length === 0) {
+    return (
+      <Card className="grid gap-6 mt-10 lg:mt-0">
+        <span className="text-2xl font-bold text-gray-900">Registration</span>
+        <span>
+          {isManager
+            ? 'Your Checkout URL assigned to this event is deleted or not valid. Please make sure assign an existing one.'
```
Comment (julien51) [inline unlock-app/src/components/content/event/EventDetails.tsx:225]:
```suggestion
            ? 'The checkout URL assigned to this event is deleted or invalid. Please make sure to assign an existing one.'
```

File: unlock-app/src/components/content/event/EventDetails.tsx (hunk 2 above)
Comment (julien51) [inline unlock-app/src/components/content/event/EventDetails.tsx:226]:
```suggestion
            : 'Registration details are not configured.'}
```

**Review by julien51 — APPROVED** [review]

File: unlock-app/src/components/content/event/EventCheckoutUrl.tsx [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+          value={settings?.slug}
+          disabled={hasCustomUrl || disabled}
+          type="text"
+          label="Custom URL"
+          error={errors?.slug?.message as string}
+          description="Custom URL that will be used for the page."
+        />
+        <div className="w-full">
+          <ToggleSwitch
+            title="Set Checkout URL."
```
Comment (kalidiagne) [inline unlock-app/src/components/content/event/EventCheckoutUrl.tsx:177]:
```suggestion
            title="Set Checkout URL"
```

**Review by julien51 — CHANGES_REQUESTED** [review]
Please add back verifiers for the locks. Just list all the locks and the corresponding verifier.

**Review by julien51 — APPROVED** [review]

**Pull request merged**